plugins/amazonq/chat/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevControllerTest.kt

```diff
@@ -390,4 +394,119 @@ class FeatureDevControllerTest : FeatureDevTestBase() {
         newFileContentsCopy[0].rejected = !newFileContentsCopy[0].rejected
         coVerify { messenger.updateFileComponent(testTabId, newFileContentsCopy, deletedFiles, "") }
     }
+
+    @Test
+    fun `test modifyDefaultSourceFolder customer does not select a folder`() = runTest {
+        val followUp = FollowUp(FollowUpTypes.MODIFY_DEFAULT_SOURCE_FOLDER, pillText = "Modify default source folder")
+        val message = IncomingFeatureDevMessage.FollowupClicked(followUp, testTabId, "", "test-command")
+
+        whenever(featureDevClient.createTaskAssistConversation()).thenReturn(exampleCreateTaskAssistConversationResponse)
+        whenever(chatSessionStorage.getSession(any(), any())).thenReturn(spySession)
+
+        mockkObject(AmazonqTelemetry)
```

Comment:
We should keep the telemetry portion separate from the implementation, not sure why we are mocking telemetry here?

Reply:
The telemetry sends network events, so mocking to me felt right to verify the object's function `modifySourceFolder` was being called with the expected arguments.

Reply:
The telemetry call here is a no-op in the test and anyway telemetry calls are automatically skipped for tests so we needn't mock telemetry here

Reply:
I'm removing the verification